File: UI/frontend/public/sigint_parser.lua
local lfs = require("lfs")

-- ===== CONFIG =====
local local_folder = "LOCALFILENAME HERE"   -- change 
local remote_user = "team19"
local remote_host = "192.168.1.123"
local remote_dir  = "/home/team19/scan"


-- ===== helpers =====
local function run_rsync()
    –print("Running rsync...")
    local ok, reason, code = os.execute(rsync_cmd)
    –print("rsync result:", ok, reason, code)
end

local function read_file(path)
    local f = io.open(path, "r")
    if not f then
        return nil
    end
    local content = f:read("*all")
    f:close()
    return content
end

local function process_file(path)
    local content = read_file(path)
    if not content then
        –print("Could not read:", path)
        return
    end

    –print("Processing:", path)
    –print(content)

    -- Put your real parsing/processing code here
end

while true do
    run_rsync()

    for file in lfs.dir(localfolder) do
        if file:match("^Scan%d+Mic%d+_output%.txt$") then
            local full = local_folder .. "/" .. file
            if not seen[full] then
                seen[full] = true
                process_file(full)
            end
        end
    end

    -- sleep ~0.75 second
    if package.config:sub(1,1) == "\" then
        os.execute("ping -n 2 127.0.0.1 > nul")
    else
        os.execute("sleep 0.75")
    end
end

File: UI/frontend/public/sigint_parser.py
import os
import re
import json
import subprocess
import time

# ===== CONFIG =====
LOCAL_FOLDER = "./data"   # MUST match your web server folder
REMOTE_USER = "team19"
REMOTE_HOST = "192.168.1.123"
REMOTE_DIR  = "/home/team19/scan"

RSYNC_CMD = [
    "rsync",
    "-avz",
    f"{REMOTE_USER}@{REMOTE_HOST}:{REMOTE_DIR}/",
    f"{LOCAL_FOLDER}/"
]

# ===== HELPERS =====
def run_rsync():
    print("Running rsync...")
    subprocess.run(RSYNC_CMD)

def read_file(path):
    try:
        with open(path, "r") as f:
            return f.read()
    except Exception as e:
        print(f"Could not read {path}: {e}")
        return None

# ===== PARSER (USES re VALUES) =====
def parse_file(path):
    content = read_file(path)
    if content is None:
        return None

    points = []
    for line in content.splitlines():
        # Match: bin ####: re=VALUE
        match = re.match(r"bin\s+(\d+):\s+re=([-\d.]+)", line)
        if match:
            bin_num, re_val = match.groups()
            points.append({
                "bin": int(bin_num),
                "re": float(re_val)
            })

    return {
        "file": os.path.basename(path),
        "points": points
    }

# ===== MAIN LOOP =====
while True:
    run_rsync()

    all_data = []

    for file_name in os.listdir(LOCAL_FOLDER):
        if re.match(r"^Scan_\d+_Mic_\d+_output\.txt$", file_name):
            full_path = os.path.join(LOCAL_FOLDER, file_name)
            parsed = parse_file(full_path)
            if parsed:
                all_data.append(parsed)

    # ===== WRITE JSON FOR FRONTEND =====
    output_path = os.path.join(LOCAL_FOLDER, "data.json")
    try:
        with open(output_path, "w") as f:
            json.dump(all_data, f, indent=4)
        print("Updated data.json")
    except Exception as e:
        print(f"Failed to write data.json: {e}")

    # ===== SLEEP (~1 second) =====
    time.sleep(1)